### PR TITLE
Homepage categories

### PR DESCRIPTION
**Pull request by danrabbit**
Reduces diff for Gtk4 a bit:

* Don't add an empty image widget to category cards with no icon
* Use calc to avoid rounding errors
* Set card padding and margin in CSS to make our lives a little simpler

### CODE REVIEW

File: data/styles/categories.css [hunk 1]
```diff
@@ -27,6 +27,9 @@
         0 3px 5px alpha(black, 0.1);
     font-size: 32px;
     font-weight: 300;
+    margin: 1.333rem; /* 12px */
+    padding: 1.333rem;
+    min-height: 3em;
```
Comment (jeremypw) [inline data/styles/categories.css:32]: You changed ems to px elsewhere but used ems here - is that OK?

File: src/Views/Homepage.vala [hunk 2]
```diff
@@ -416,38 +407,33 @@ public class AppCenter.Homepage : Gtk.Box {
...
             };
-            box.add (display_image);
-            box.add (name_label);
 
             content_area.attach (box, 0, 0);
 
             if (category.icon != "") {
-                display_image.icon_name = category.icon;
+                var display_image = new Gtk.Image.from_icon_name (category.icon, Gtk.IconSize.DIALOG) {
+                     halign = Gtk.Align.END,
```
Comment (jeremypw) [inline src/Views/Homepage.vala:424]: Incorrect indent.

**Review by jeremypw — COMMENTED** [review]
No regression in appearance.  Just a couple of minor comments.

File: data/styles/categories.css (hunk 1 above)
Comment (danrabbit) [reply data/styles/categories.css:32]: Yeah it works for margins and padding, but when used as stops in gradients we get weird rounding sometimes that makes gradients either too hard or lines disappear etc

**Comment by danrabbit** [comment]
After thinking about it for a second, we can use `calc` to avoid rounding errors and keep the background image scalable :tada:

**Review by jeremypw — APPROVED** [review]

**Pull request merged**